src/awkward/_parameters.py

```diff
@@ -81,21 +81,21 @@ def parameters_intersect(
     left: JSONMapping | None,
     right: JSONMapping | None,
     *,
-    exclude: Collection[tuple[str, JSONSerializable]] = (),
+    exclude: Collection[tuple[str, Any]] | None = None,
 ) -> JSONMapping | None:
     """
     Args:
         left: first parameters mapping
         right: second parameters mapping
-        exclude: collection of (key, value) items to exclude
+        exclude: collection of keys to exclude
 
     Returns the intersected key-value pairs of `left` and `right` as a dictionary.
     """
     if left is None or right is None:
         return None
 
     common_keys = iter(left.keys() & right.keys())
-    has_no_exclusions = len(exclude) == 0
+    has_exclusions = bool(exclude)
```

Comment:
`bool(x)` is a rather brief way to say `len(x) != 0`. It works through Python's truthiness, but I would prefer it if Python required us to be more explicit. Nevertheless, this does look like something a flake8-style linter would suggest. `:)`

Reply:
Hehe, I wondered if you'd mind this given that linters generally prefer it. Here, I want to mean both `x is not None` or `len(x) != 0`

Reply:
I prefer `x is not None` and also `len(x) != 0` over `bool(x)`, but now that we're talking

```python
x is not None and len(x) != 0
```

Well, that is a lot more characters than

```python
bool(x)
```

~~I guess it's okay.~~ You know, the fact that I didn't realize that `x` might be `None` by reading the code is an argument in favor of the long expression. I know that what type hints are for, but the type hint is far away and long-time Python users have gotten used to looking for clues about a variable's type from nearby context. So I still kinda prefer the long expression, as long as black doesn't split it up onto multiple lines. (I don't think it would.)